packages/wallet/src/services/UtxoTracker.ts

```diff
@@ -119,13 +119,20 @@ export const createUtxoTracker = (
 
   return {
     available$,
-    setUnspendable: unspendableUtxoSource$.next.bind(unspendableUtxoSource$),
+    setUnspendable: async (utxo) => {
+      unspendableUtxoSource$.next(utxo);
+    },
     shutdown: () => {
       utxoSource$.complete();
       unspendableUtxoSource$.complete();
       logger.debug('Shutdown');
     },
     total$,
-    unspendable$: unspendableUtxoSource$
+    unspendable$: combineLatest([unspendableUtxoSource$, total$]).pipe(
+      map(([unspendableUtxo, utxo]) =>
+        utxo.filter(([utxoTxIn]) => unspendableUtxo.some(([unspendable]) => txInEquals(utxoTxIn, unspendable)))
```

Comment:
There will be less `unspendableUtxo` than `total`, so this would be slightly more efficient
```suggestion
        unspendableUtxo.filter(([unspendable]) => utxo.some(([utxoTxIn]) => txInEquals(utxoTxIn, unspendable)))
```

Reply:
Done in https://github.com/input-output-hk/cardano-js-sdk/pull/605/commits/e6007465e617698954a61774b3c2a461678c322a (squashed due to time constraints)